### DISCUSSION TITLE
linux环境tbox项目build ios时报错

### DISCUSSION THREAD

**Discussion by jmgaooo** [xmake-io/xmake#3090] [Q&A]
问题：第一次用xmake，文档没看到具体 的案例，不知道是否这样构建ios项目，麻烦指导一下。目标是构建一个ios的可执行文件或者库文件

操作：
xmake create -l c -t console_tbox test
cd test
xmake f -v -p iphoneos -a arm64 -m debug

输出：
checking for Xcode directory ... %s
checking for Codesign Identity of Xcode ... no
checking for Mobile Provision of Xcode ... no
checking for unzip ... /usr/bin/unzip
checking for git ... /usr/bin/git
checking for gzip ... /usr/bin/gzip
checking for tar ... /usr/bin/tar
/usr/bin/git rev-parse HEAD
checking for xmake::tbox ... no
note: install or modify (m) these packages (pass -y to skip confirm)?
in xmake-repo:
  -> tbox v1.7.1 [debug:y]
please input: y (y/n/m)
y
checking for ping ... /usr/bin/ping
pinging for the host(gitee.com) ... 10 ms
pinging for the host(gitlab.com) ... 200 ms
pinging for the host(github.com) ... 65535 ms
/usr/bin/tar -xf v1.7.1.tar.gz -C source.tmp
xmake f --verbose -y -c --plat=iphoneos --arch=arm64 --mode=debug --kind=static --cxflags=-fPIC --demo=false
checking for Xcode directory ... %s
checking for Codesign Identity of Xcode ... no
checking for Mobile Provision of Xcode ... no
> checking for c includes(sched.h)
> checking for c funcs(sched_setaffinity)
checking for posix_sched_setaffinity ... no
> checking for c includes(math.h)
> checking for c funcs(exp)
checking for libm_exp ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcsncasecmp)
checking for libc_wcsncasecmp ... no
> checking for c includes(sched.h)
> checking for c funcs(sched_yield)
checking for posix_sched_yield ... no
> checking for c includes(stdio.h)
> checking for c funcs(fgets)
checking for libc_fgets ... no
> checking for c includes(sys/stat.h)
> checking for c funcs(futimens)
checking for posix_futimens ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcscmp)
checking for libc_wcscmp ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcscasestr)
checking for libc_wcscasestr ... no
> checking for c includes(math.h)
> checking for c funcs(sincos)
checking for libm_sincos ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strlcpy)
checking for libc_strlcpy ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcsnlen)
checking for libc_wcsnlen ... no
> checking for c includes(sys/sem.h, sys/ipc.h)
> checking for c funcs(semtimedop)
checking for systemv_semtimedop ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_setaffinity_np)
checking for posix_pthread_setaffinity_np ... no
> checking for c includes(math.h)
> checking for c funcs(sinf)
checking for libm_sinf ... no
> checking for c includes(unistd.h)
> checking for c funcs(pread64)
checking for posix_pread64 ... no
> checking for c includes(time.h)
> checking for c funcs(gmtime)
checking for libc_gmtime ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcscasecmp)
checking for libc_wcscasecmp ... no
> checking for c includes(stdio.h)
> checking for c funcs(fwrite)
checking for libc_fwrite ... no
> checking for c includes(sys/socket.h, fcntl.h)
> checking for c funcs(socket)
checking for posix_socket ... no
> checking for c includes(semaphore.h)
> checking for c funcs(sem_init)
checking for posix_sem_init ... no
> checking for c includes(stdlib.h)
> checking for c funcs(random)
checking for libc_random ... no
> checking for c includes(math.h)
> checking for c funcs(tan)
checking for libm_tan ... no
> checking for c includes(stdio.h)
> checking for c funcs(ungetc)
checking for libc_ungetc ... no
> checking for c includes(math.h)
> checking for c funcs(asin)
checking for libm_asin ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strncpy)
checking for libc_strncpy ... no
> checking for c includes(netdb.h)
> checking for c funcs(getaddrinfo)
checking for posix_getaddrinfo ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_getspecific)
checking for posix_pthread_getspecific ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strnlen)
checking for libc_strnlen ... no
> checking for c includes(netdb.h)
> checking for c funcs(gethostbyaddr)
checking for posix_gethostbyaddr ... no
> checking for c includes(math.h)
> checking for c funcs(log2)
checking for libm_log2 ... no
> checking for c includes(netdb.h)
> checking for c funcs(getnameinfo)
checking for posix_getnameinfo ... no
> checking for c includes(stdio.h)
> checking for c funcs(fputc)
checking for libc_fputc ... no
> checking for c includes(signal.h, setjmp.h)
> checking for c funcs(kill)
checking for libc_kill ... no
> checking for c includes(sys/select.h)
> checking for c funcs(select)
checking for posix_select ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strchr)
checking for libc_strchr ... no
> checking for c includes(signal.h, setjmp.h)
> checking for c funcs(signal)
checking for libc_signal ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcscat)
checking for libc_wcscat ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_key_delete)
checking for posix_pthread_key_delete ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(memmem)
checking for libc_memmem ... no
> checking for c includes(locale.h)
> checking for c funcs(setlocale)
checking for libc_setlocale ... no
> checking for c includes(math.h)
> checking for c funcs(cos)
checking for libm_cos ... no
> checking for c includes(sys/stat.h)
> checking for c funcs(mkfifo)
checking for posix_mkfifo ... no
> checking for c includes(math.h)
> checking for c funcs(sqrtf)
checking for libm_sqrtf ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcsncpy)
checking for libc_wcsncpy ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strstr)
checking for libc_strstr ... no
> checking for c includes(sys/sendfile.h)
> checking for c funcs(sendfile)
checking for posix_sendfile ... no
> checking for c includes(math.h)
> checking for c funcs(cosf)
checking for libm_cosf ... no
> checking for c includes(sys/stat.h, fcntl.h)
> checking for c funcs(lstat64)
checking for posix_lstat64 ... no
> checking for c includes(math.h)
> checking for c funcs(powf)
checking for libm_powf ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcsncmp)
checking for libc_wcsncmp ... no
> checking for c includes(unistd.h)
> checking for c funcs(sysconf)
checking for posix_sysconf ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strrchr)
checking for libc_strrchr ... no
> checking for c includes(execinfo.h)
> checking for c funcs(backtrace)
checking for libc_backtrace ... no
> checking for c includes(math.h)
> checking for c funcs(fmodf)
checking for libm_fmodf ... no
> checking for c includes(unistd.h)
> checking for c funcs(gethostname)
checking for posix_gethostname ... no
> checking for c includes(stdio.h)
> checking for c funcs(fputs)
checking for libc_fputs ... no
> checking for c includes(netdb.h)
> checking for c funcs(gethostbyname)
checking for posix_gethostbyname ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strcasestr)
checking for libc_strcasestr ... no
> checking for c includes(unistd.h)
> checking for c funcs(getpagesize)
checking for posix_getpagesize ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(memcmp)
checking for libc_memcmp ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcslen)
checking for libc_wcslen ... no
> checking for c includes(math.h)
> checking for c funcs(atan2)
checking for libm_atan2 ... no
> checking for c includes(unistd.h)
> checking for c funcs(execvpe)
checking for posix_execvpe ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strncasecmp)
checking for libc_strncasecmp ... no
> checking for c includes(sys/epoll.h)
> checking for c funcs(epoll_wait)
checking for posix_epoll_wait ... no
> checking for c includes(dirent.h)
> checking for c funcs(opendir)
checking for posix_opendir ... no
> checking for c includes(sys/stat.h)
> checking for c funcs(utimensat)
checking for posix_utimensat ... no
> checking for c includes(spawn.h)
> checking for c funcs(posix_spawnp)
checking for posix_posix_spawnp ... no
> checking for c includes(unistd.h)
> checking for c funcs(getdtablesize)
checking for posix_getdtablesize ... no
> checking for c includes(signal.h, setjmp.h)
> checking for c funcs(setjmp)
checking for libc_setjmp ... no
> checking for c includes(spawn.h)
> checking for c funcs(posix_spawn_file_actions_addchdir_np)
checking for posix_posix_spawn_file_actions_addchdir_np ... no
> checking for c includes(math.h)
> checking for c funcs(asinf)
checking for libm_asinf ... no
> checking for c includes(sys/stat.h, fcntl.h)
> checking for c funcs(stat64)
checking for posix_stat64 ... no
> checking for c includes(sys/file.h, fcntl.h)
> checking for c funcs(flock)
checking for bsd_flock ... no
> checking for c includes(sys/mman.h)
> checking for c funcs(mmap)
checking for posix_mmap ... no
> checking for c includes(sys/uio.h)
> checking for c funcs(preadv)
checking for posix_preadv ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_mutex_init)
checking for posix_pthread_mutex_init ... no
> checking for c includes(unistd.h)
> checking for c funcs(pipe)
checking for posix_pipe ... no
> checking for c includes(stdio.h)
> checking for c funcs(fgetc)
checking for libc_fgetc ... no
> checking for c links(pthread)
> checking for c snippet(TB_CONFIG_KEYWORD_HAVE_Thread_local)
checking for keyword_thread_local ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strncmp)
checking for libc_strncmp ... no
> checking for c includes(math.h)
> checking for c funcs(expf)
checking for libm_expf ... no
> checking for c includes(math.h)
> checking for c funcs(atanf)
checking for libm_atanf ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcscpy)
checking for libc_wcscpy ... no
> checking for c includes(regex.h)
> checking for c funcs(regcomp)
checking for posix_regcomp ... no
> checking for c includes(sys/uio.h)
> checking for c funcs(pwritev)
checking for posix_pwritev ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strcat)
checking for libc_strcat ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcsncat)
checking for libc_wcsncat ... no
> checking for c includes(sys/epoll.h)
> checking for c funcs(epoll_create)
checking for posix_epoll_create ... no
> checking for c includes(unistd.h)
> checking for c funcs(fork)
checking for posix_fork ... no
> checking for c includes(regex.h)
> checking for c funcs(regexec)
checking for posix_regexec ... no
> checking for c includes(math.h)
> checking for c funcs(sincosf)
checking for libm_sincosf ... no
> checking for c includes(unistd.h)
> checking for c funcs(pipe2)
checking for posix_pipe2 ... no
> checking for c includes(math.h)
> checking for c funcs(atan)
checking for libm_atan ... no
> checking for c includes(math.h)
> checking for c funcs(atan2f)
checking for libm_atan2f ... no
> checking for c includes(unistd.h)
> checking for c funcs(fdatasync)
checking for posix_fdatasync ... no
> checking for c includes(math.h)
> checking for c funcs(fmod)
checking for libm_fmod ... no
> checking for c includes(poll.h, sys/socket.h)
> checking for c funcs(poll)
checking for posix_poll ... no
> checking for c includes(unistd.h)
> checking for c funcs(vfork)
checking for posix_vfork ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strcasecmp)
checking for libc_strcasecmp ... no
> checking for c includes(valgrind/valgrind.h)
> checking for c funcs(VALGRIND_STACK_REGISTER)
checking for valgrind_VALGRIND_STACK_REGISTER ... no
> checking for c includes(math.h)
> checking for c funcs(tanf)
checking for libm_tanf ... no
> checking for c includes(sys/uio.h)
> checking for c funcs(readv)
checking for posix_readv ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(mbstowcs)
checking for libc_mbstowcs ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(memset)
checking for libc_memset ... no
> checking for c links(pthread)
> checking for c snippet(TB_CONFIG_KEYWORD_HAVE__thread)
checking for keyword_thread ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strncat)
checking for libc_strncat ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strlen)
checking for libc_strlen ... no
> checking for c includes(ifaddrs.h)
> checking for c funcs(getifaddrs)
checking for posix_getifaddrs ... no
> checking for c includes(stdlib.h)
> checking for c funcs(srandom)
checking for libc_srandom ... no
> checking for c includes(sys/uio.h)
> checking for c funcs(writev)
checking for posix_writev ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_key_create)
checking for posix_pthread_key_create ... no
> checking for c includes(sys/time.h)
> checking for c funcs(gettimeofday)
checking for libc_gettimeofday ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(memmove)
checking for libc_memmove ... no
> checking for c includes(fcntl.h)
> checking for c funcs(fcntl)
checking for posix_fcntl ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcstombs)
checking for libc_wcstombs ... no
> checking for c includes(time.h)
> checking for c funcs(localtime)
checking for libc_localtime ... no
> checking for c includes(math.h)
> checking for c funcs(acosf)
checking for libm_acosf ... no
> checking for c includes(sys/resource.h)
> checking for c funcs(getrlimit)
checking for posix_getrlimit ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strcmp)
checking for libc_strcmp ... no
> checking for c snippet(TB_CONFIG_FEATURE_HAVE_ANONYMOUS_UNION)
checking for feature_anonymous_union ... no
> checking for c includes(sys/stat.h, fcntl.h)
> checking for c funcs(open)
checking for posix_open ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcslcpy)
checking for libc_wcslcpy ... no
> checking for c includes(unistd.h)
> checking for c funcs(pwrite64)
checking for posix_pwrite64 ... no
> checking for c includes(dlfcn.h)
> checking for c funcs(dlopen)
checking for posix_dlopen ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(strcpy)
checking for libc_strcpy ... no
> checking for c includes(math.h)
> checking for c funcs(sin)
checking for libm_sin ... no
> checking for c includes(sys/wait.h)
> checking for c funcs(waitpid)
checking for posix_waitpid ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_create)
checking for posix_pthread_create ... no
> checking for c includes(string.h, stdlib.h)
> checking for c funcs(memcpy)
checking for libc_memcpy ... no
> checking for c includes(wchar.h, stdlib.h)
> checking for c funcs(wcsstr)
checking for libc_wcsstr ... no
> checking for c includes(math.h)
> checking for c funcs(acos)
checking for libm_acos ... no
> checking for c includes(math.h)
> checking for c funcs(log2f)
checking for libm_log2f ... no
> checking for c includes(time.h)
> checking for c funcs(mktime)
checking for libc_mktime ... no
> checking for c includes(signal.h, setjmp.h)
> checking for c funcs(sigsetjmp)
checking for libc_sigsetjmp ... no
> checking for c includes(math.h)
> checking for c funcs(pow)
checking for libm_pow ... no
> checking for c includes(pthread.h)
> checking for c funcs(pthread_setspecific)
checking for posix_pthread_setspecific ... no
> checking for c includes(unistd.h)
> checking for c funcs(execvp)
checking for posix_execvp ... no
> checking for c includes(stdio.h)
> checking for c funcs(fread)
checking for libc_fread ... no
> checking for c includes(sys/sem.h, sys/ipc.h)
> checking for c funcs(semget)
checking for systemv_semget ... no
> checking for c types(wchar_t)
checking for wchar ... no
> checking for c includes(copyfile.h)
> checking for c funcs(copyfile)
checking for posix_copyfile ... no
> checking for c includes(math.h)
> checking for c funcs(sqrt)
checking for libm_sqrt ... no
generating src/tbox/tbox.config.h.in to build/iphoneos/arm64/debug/tbox.config.h ..
checking for git ... /usr/bin/git
generating src/tbox/tbox.config.h.in ... ok
error: attempt to concatenate a nil value (local 'target_minver')
error: execv(xmake f --verbose -y -c --plat=iphoneos --arch=arm64 --mode=debug --kind=static --cxflags=-fPIC --demo=false) failed(255)
  => install tbox v1.7.1 .. failed
error: install failed!

**Comment by waruqi** [comment]
提供 xmake 版本，以及带 -vD 的详细完整log

```console
xmake --version
xmake f -p iphoneos -m debug -cvD
xmake -rvD
```

**Reply by waruqi** [reply]
另外，问题反馈，走 issues

**Reply by jmgaooo** [reply]
> 另外，问题反馈，走 issues

谢谢，已转到issues，补充了版本和详细的日志，在这里 #3091
